web/src/topic_popover.ts

```diff
@@ -27,20 +27,18 @@ function get_conversation(instance: tippy.Instance): {
 } {
     let stream_id;
     let topic_name;
-    let url;
 
     if (!instance.reference.classList.contains("topic-sidebar-menu-icon")) {
         const $elt = $(instance.reference);
         stream_id = Number.parseInt($elt.attr("data-stream-id")!, 10);
         topic_name = $elt.attr("data-topic-name")!;
-        url = new URL($elt.attr("data-topic-url")!, realm.realm_url).href;
     } else {
         const $elt = $(instance.reference).closest(".topic-sidebar-menu-icon").expectOne();
         const $stream_li = $elt.closest(".narrow-filter").expectOne();
         topic_name = $elt.closest("li").expectOne().attr("data-topic-name")!;
-        url = util.the($elt.closest("li").find<HTMLAnchorElement>("a.sidebar-topic-name")).href;
         stream_id = stream_popover.elem_to_stream_id($stream_li);
     }
+    const url = hash_util.by_channel_topic_permalink(stream_id, topic_name);
```

Comment:
The `new URL` pattern is needed here, otherwise you're copying just an `#narrow` bit, apparently.

Reply:
Updated this.